src/scheduler_utils.cc

```diff
@@ -33,6 +33,46 @@
 
 namespace triton { namespace core {
 
+uint64_t
+CaptureTimeNs()
+{
+  return std::chrono::duration_cast<std::chrono::nanoseconds>(
+             std::chrono::steady_clock::now().time_since_epoch())
+      .count();
+}
+
+bool
+CacheLookUpUtil(
+    std::unique_ptr<InferenceRequest>& request,
+    std::unique_ptr<InferenceResponse>& cached_response,
+    std::shared_ptr<TritonCache> cache)
+{
+  Status status;
+  std::unique_ptr<InferenceResponse> local_response;
+  request->ResponseFactory()->CreateResponse(&local_response);
+  std::string key = "";
+  if (!request->CacheKeyIsSet()) {
+    status = cache->Hash(*request, &key);
+    if (!status.IsOk()) {
+      LOG_ERROR << "Failed to hash request: " << status.Message();
+      return false;
+    }
+    request->SetCacheKey(key);
+  } else {
+    key = request->CacheKey();
+  }
+  {
+    request->CaptureCacheLookupStartNs();
+    status = cache->Lookup(local_response.get(), key);
+    request->CaptureCacheLookupEndNs();
+  }
```

Comment:
Why this needs to be in a separate block?

Reply:
I did not implement this function. It was written by the someone who implemented CacheLookup() function in dynamic_batcher.cc. I only moved it to a separate function in scheduler_utils.cc so that it can be reused by both dynamic scheduler and ensemble schedulers.

Reply:
Probably just a logical block, no functional issues. I don't mind removing it.

Reply:
Up to you and @lkomali, I was just curious if I'm missing something